### DISCUSSION TITLE
Can i run in docker swarm

### DISCUSSION THREAD

**Discussion by timeshift92** [microsoft/aspire#4239] [Q&A]
Hello good instrument. but in my work. we use docker swarm for orchestration, how can i use aspire in swarm mode?

**Comment by fkornel** [comment]
Interesting question. upup.

**Comment by vikto132** [comment]
up

**Comment by davidfowl** [comment]
Are you asking about deployment or development?

**Reply by jjchiw** [reply]
I think it's deployment.

I know two PAAS open source that uses docker swarm
- https://dokploy.com/
- https://caprover.com/

And sure it would be awesome to `aspire deploy` to one of those PAAS

I was looking into the `aspire` deployment to docker-compose and I saw that there are some stuff there `https://github.com/dotnet/aspire/tree/main/src/Aspire.Hosting.Docker/Resources/ServiceNodes/Swarm` I could copy the output and paste it in dokploy

or maybe it would be easier to just generate the images push the images to repository
create the services like (postgres, redis)
wire everything up as env variables
and use the dokploy api https://docs.dokploy.com/docs/api/application#application-save-docker-provider

Ok. I have a plan. :)

**Reply by jjchiw** [reply]
and the plan is "done" :)

https://github.com/Cosmic-Chimps/aspire-hosting-dokploy

still in preview, but it's working for me :)

**Reply by davidfowl** [reply]
".NET Aspire" 😢

**Reply by jjchiw** [reply]
Done
Just `Aspire` 
:)
